rules/manager.go

```diff
@@ -743,6 +743,47 @@ func (g *Group) RestoreForState(ts time.Time) {
 
 }
 
+// Equals return if two groups have the same
+// 1. name
+// 2. file path
+// 3. interval
+// 4. rule set. and the rules are the same
+func (g *Group) Equals(ng *Group) bool {
+	if g.name != ng.name {
+		return false
+	}
+
+	if g.file != ng.file {
+		return false
+	}
+
+	if g.interval != ng.interval {
+		return false
+	}
+
+	if len(g.rules) != len(ng.rules) {
+		return false
+	}
+
+	gm := make(map[string]Rule, len(g.rules))
+	for _, r := range g.rules {
+		gm[r.String()] = r
```

Comment:
why not just using a slice? and them compare them one by once? changing the order of rules is a different group.

Reply:
So the order of the rules also matters, right?

Reply:
That would make things much easier

Reply:
done.